subgraphs/burrow/src/handlers/actions.ts

```diff
@@ -0,0 +1,589 @@
+
+  const withdraw = getOrCreateWithdrawal(
+    receipt.outcome.id
+      .toBase58()
+      .concat("-")
+      .concat((logIndex as i32).toString()),
+    receipt
+  );
+  withdraw.logIndex = logIndex as i32;
+  const parsedData = parse0(data);
+  const account_id = parsedData[0];
+  const amount = parsedData[1];
+  const token_id = parsedData[2];
+
+  const market = getOrCreateMarket(token_id.toString());
+  const dailySnapshot = getOrCreateMarketDailySnapshot(market, receipt);
+  const hourlySnapshot = getOrCreateMarketHourlySnapshot(market, receipt);
+  const usageDailySnapshot = getOrCreateUsageMetricsDailySnapshot(receipt);
+  const usageHourlySnapshot = getOrCreateUsageMetricsHourlySnapshot(receipt);
+  const financialDailySnapshot = getOrCreateFinancialDailySnapshot(receipt);
+  const account = getOrCreateAccount(account_id.toString());
+  const position = getOrCreatePosition(account, market, "LENDER", receipt);
+
+  const token = getOrCreateToken(token_id.toString());
+  withdraw.account = account.id;
+  withdraw.amount = BigInt.fromString(amount.toString());
+  withdraw.asset = token.id;
+  withdraw.market = market.id;
+  withdraw.position = position.id;
+
+  withdraw.amountUSD = withdraw.amount
+    .divDecimal(
+      BigInt.fromI32(10)
+        .pow((token.decimals + token.extraDecimals) as u8)
+        .toBigDecimal()
+    )
+    .times(token.lastPriceUSD!);
+
+  // usage metrics
+  if (account._last_active_timestamp.lt(usageDailySnapshot.timestamp)) {
+    usageDailySnapshot.dailyActiveUsers += 1;
+    usageHourlySnapshot.hourlyActiveUsers += 1;
+  }
+  usageDailySnapshot.dailyWithdrawCount += 1;
+  usageDailySnapshot.dailyTransactionCount += 1;
+  usageHourlySnapshot.hourlyWithdrawCount += 1;
+  usageHourlySnapshot.hourlyTransactionCount += 1;
+
+  position.withdrawCount += 1;
+
+  // close if balance is zero
+  if (position.balance.lt(withdraw.amount)) {
+    position.balance = BI_ZERO;
+    account.openPositionCount -= 1;
+    account.closedPositionCount += 1;
+
+    market.openPositionCount -= 1;
+    market.closedPositionCount += 1;
+    market.lendingPositionCount -= 1;
+
+    position.hashClosed = receipt.outcome.id.toBase58();
+    position.timestampClosed = BigInt.fromU64(
+      receipt.block.header.timestampNanosec / 1000000000
+    );
+    position.blockNumberClosed = BigInt.fromU64(receipt.block.header.height);
+  }
+
+  account.withdrawCount = account.withdrawCount + 1;
+  market._totalWithrawnHistory = market._totalWithrawnHistory.plus(
+    withdraw.amount
+  );
+
+  market.outputTokenSupply = market.outputTokenSupply.minus(
+    amount_to_shares(
+      BI_BD(withdraw.amount),
+      market.outputTokenSupply,
+      market._totalDeposited
+    )
+  );
+  market._totalDeposited = market._totalDeposited.minus(BI_BD(withdraw.amount));
+
+  // if total deposited is negative, we need to move the negative amount to the reserve
+  // this can [possibly] happen if amount is withdrawn from reserve
```

Comment:
Is this an attempt to catch that missed balance from not being able to account for interest?